website/content/en/docs/building-operators/golang/quickstart.md

```diff
@@ -1,500 +1,102 @@
-
-
-#### Reconcile loop
-
-Every Controller has a Reconciler object with a `Reconcile()` method that implements the reconcile loop. The reconcile loop is passed the [`Request`][request-go-doc] argument which is a Namespace/Name key used to lookup the primary resource object, Memcached, from the cache:
-
-```Go
-import (
-	ctrl "sigs.k8s.io/controller-runtime"
-
-	cachev1alpha1 "github.com/example-inc/memcached-operator/api/v1alpha1"
-	...
-)
-
-func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-  // Lookup the Memcached instance for this reconcile request
-  memcached := &cachev1alpha1.Memcached{}
-  err := r.Get(ctx, req.NamespacedName, memcached)
-  ...
-}
-```
-
-Based on the return values, [`Result`][result_go_doc] and error, the `Request` may be requeued and the reconcile loop may be triggered again:
-
-```Go
-// Reconcile successful - don't requeue
-return ctrl.Result{}, nil
-// Reconcile failed due to error - requeue
-return ctrl.Result{}, err
-// Requeue for any reason other than an error
-return ctrl.Result{Requeue: true}, nil
-```
-
-You can set the `Result.RequeueAfter` to requeue the `Request` after a grace period as well:
-```Go
-import "time"
-
-// Reconcile for any reason other than an error after 5 seconds
-return ctrl.Result{RequeueAfter: time.Second*5}, nil
-```
-
-**Note:** Returning `Result` with `RequeueAfter` set is how you can periodically reconcile a CR.
-
-For a guide on Reconcilers, Clients, and interacting with resource Events, see the [Client API doc][doc_client_api].
-
-### Specify permissions and generate RBAC manifests
-
-The controller needs certain RBAC permissions to interact with the resources it manages. These are specified via [RBAC markers][rbac_markers] like the following:
-
-```Go
-// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds/status,verbs=get;update;patch
-// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;
-
-func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-```
-
-The `ClusterRole` manifest at `config/rbac/role.yaml` is generated from the above markers via controller-gen with the following command:
-
-```sh
-$ make manifests
-```
-
-## Build and run the operator
-
-Before running the operator, the CRD must be registered with the Kubernetes apiserver:
+Projects are scaffolded with tests that requires certain Kubernetes server binaries be present locally:
 
 ```sh
-$ make install
+curl -sSLo setup_envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/kubebuilder/master/scripts/setup_envtest_bins.sh 
+chmod +x setup_envtest.sh
+./setup_envtest.sh v1.18.2 v3.4.3
 ```
 
-Once this is done, there are two ways to run the operator:
+**Note:** More info can be found [here][env-test-setup].
```

Comment:
We need to update this page based on kubernetes-sigs/kubebuilder#1615 as well.

Can be done in a follow-up.

Reply:
Hi @joelanford, 

The envtest doc update was done in: https://github.com/operator-framework/operator-sdk/pull/3543
We cannot say here to use the new target anyway because it will be done just for v3 if the PR get merged and we are using v2.